TypeCobol/Compiler/Parser/CodeElementBuilder/CobolStatementsBuilder.cs

```diff
@@ -744,23 +744,98 @@ internal InvokeStatement CreateInvokeStatement(CodeElementsParser.InvokeStatemen
+            var jsonNameMapping = new JsonNameMapping
+            {
+                DataItem = CobolExpressionsBuilder.CreateVariable(context.dataItem),
+                OutputName = CobolWordsBuilder.CreateAlphanumericValue(context.outputName)
+            };
+            if (context.OMITTED() != null)
+            {
+                jsonNameMapping.Omitted = new SyntaxProperty<bool>(true, ParseTreeUtils.GetFirstToken(context.OMITTED()));
+            }
+
+            return jsonNameMapping;
+        }
+
+        private JsonSuppressDirective CreateJsonSuppressDirective(CodeElementsParser.JsonSuppressDirectiveContext context)
+        {
+            var suppressDirective = new JsonSuppressDirective();
+            if (context.subordinateDataItem != null)
+            {
+                suppressDirective.DataItemName = CobolExpressionsBuilder.CreateVariable(context.subordinateDataItem);
+                if (context.whenPhrase() != null)
+                {
+                    suppressDirective.ItemValuesToSuppress = BuildObjectArrayFromParserRules(context.whenPhrase().repeatedCharacterValue3(), ctx => CobolWordsBuilder.CreateRepeatedCharacterValue(ctx));
+                }
+            }
+            else if (context.jsonGenericSuppressionPhrase() != null)
+            {
+                var suppression = context.jsonGenericSuppressionPhrase();
+
+                if (suppression.EVERY() != null)
+                {
+                    if (suppression.NUMERIC() != null)
+                    {
+                        suppressDirective.JsonItemTypeToSuppress = CreateSyntaxProperty(JsonItemType.NUMERIC, suppression.NUMERIC());
+                    }
+                    else if (suppression.nonnumeric() != null)
+                    {
+                        suppressDirective.JsonItemTypeToSuppress = CreateSyntaxProperty(JsonItemType.NONNUMERIC, suppression.nonnumeric().UserDefinedWord());
+                    }
+                }
+
+                if (suppression.whenPhrase() != null)
+                {
+                    suppressDirective.ItemValuesToSuppress = BuildObjectArrayFromParserRules(suppression.whenPhrase().repeatedCharacterValue3(), ctx => CobolWordsBuilder.CreateRepeatedCharacterValue(ctx));
+                }
+            }
+
+            return suppressDirective;
+        }
+
+        private JsonGenerateConvertingDirective CreateJsonConvertingDirective(CodeElementsParser.JsonGenerateConvertingDirectiveContext context)
+        {
+            var dataItem = CobolExpressionsBuilder.CreateVariable(context.convertingDataItem);
+            Variable trueValue = null;
+
+            if (context.qualifiedConditionName() != null)
+            {
+                var symbolReference = CobolWordsBuilder.CreateQualifiedConditionName(context.qualifiedConditionName());
+                StorageArea storageArea = new DataOrConditionStorageArea(symbolReference, false);
+                trueValue = new Variable(storageArea);
```

Comment:
No pre-existing method for that ? The problem here is that you don't capture the newly created StorageArea.
Each method creating a StorageArea should either add the instance to the reads/writes dictionary itself or ensure that all callers do it themselves.

Reply:
I think I had the same problem for XML GENERATE / PARSE

Reply:
Ok, to summarize:
- when going from a `StorageArea` to a `Variable`, the method responsible for the variable creation also stores the storage area into one of the two dictionaries `storageAreaReads` or `storageAreaWrites`
  - this is tricky because depending on the statement it could be a read or a write and the methods do not have enough context to decide...

For the present case, add a new method that build a variable from a (new) `ConditionVariableContext` and add the area to the reads dictonary. This will match the existing code.
For `SORT` statement, I forgot to register the file name, so could you add a boolean in `CreateDataItemReferenceOrFileName` method signature and the code to register the area ?

I think this could become the new pattern to avoid categorizing reads as writes and vice-versa, but this has to be looked more thoroughly...

Reply:
Done for JSON GENERATE / PARSE
For SORT:
No boolean is needed as `CreateDataItemReferenceOrFileName ` returns a DataOrConditionStorageArea which is implicitely a 'read' ('write' are only ReceivingStorageArea).
So capture in `storageAreaReads `is enough.